components/portfolio/BatchListings.tsx

```diff
@@ -0,0 +1,1033 @@
+    setListings((prevListings) => {
+      return prevListings.map((listing) => {
+        if (listing.token.token?.attributes) {
+          // Find the highest floor price
+          let topTraitPrice = Math.max(
+            ...listing.token.token.attributes.map(
+              (attribute) => attribute.floorAskPrice ?? 0
+            )
+          )
+          if (topTraitPrice && topTraitPrice > 0) {
+            return {
+              ...listing,
+              price: topTraitPrice.toString(),
+            }
+          }
+        }
+
+        return listing
+      })
+    })
+    setCurrency(defaultCurrency)
+  }, [listings])
+
+  return (
+    <Flex direction="column" css={{ gap: '$5', width: '100%' }}>
+      <Flex align="center" css={{ gap: 24 }}>
+        <Button
+          color="gray3"
+          size="small"
+          css={{ justifyContent: 'center', width: '44px', height: '44px' }}
+          onClick={() => setShowListingPage(false)}
+        >
+          <FontAwesomeIcon icon={faChevronLeft} />
+        </Button>
+        <Text style="h4">List for Sale</Text>
+      </Flex>
+      <Flex
+        justify="between"
+        css={{ border: '1px solid $gray6', borderRadius: 8, p: 24 }}
+      >
+        <Flex direction="column" css={{ gap: '$3' }}>
+          <Text style="h6">Select Marketplaces</Text>
+          <Flex align="center" css={{ gap: '$3' }}>
+            {marketplaces.map((marketplace) => {
+              const isSelected = selectedMarketplaces.some(
+                (selected) => selected.orderbook === marketplace.orderbook
+              )
+
+              return (
+                <Flex
+                  key={marketplace.name}
+                  align="center"
+                  css={{
+                    border: isSelected
+                      ? '1px solid $primary7'
+                      : '1px solid $gray6',
+                    borderRadius: 8,
+                    gap: '$2',
+                    p: '$3',
+                    cursor: 'pointer',
+                  }}
+                  onClick={() => handleMarketplaceSelection(marketplace)}
+                >
+                  <img
+                    src={marketplace.imageUrl}
+                    alt={marketplace.name}
+                    style={{ width: 32, height: 32 }}
+                  />
+                  <Text style="subtitle2">{marketplace.name}</Text>
+                </Flex>
+              )
+            })}
+          </Flex>
+        </Flex>
+        <Flex direction="column" css={{ gap: '$3' }}>
+          <Text style="h6">Apply to All</Text>
+          <Flex align="center" css={{ gap: '$5' }}>
+            {isLargeDevice ? (
+              <Flex align="center" css={{ gap: '$3' }}>
+                <Button
+                  color="gray3"
+                  corners="pill"
+                  size="large"
+                  css={{ minWidth: 'max-content' }}
+                  onClick={() => applyFloorPrice()}
```

Comment:
I think you can just do this right:
`onClick={applyFloorPrice}`
Same as the one below for the top trait